### PR TITLE
fix: Fix stdouttrace/example_test to make the trace_id same.

### PR DESCRIPTION
**Pull request by StLeoX**

### CODE REVIEW

File: exporters/stdout/stdouttrace/example_test.go [hunk 1]
```diff
@@ -18,12 +18,13 @@ import (
...
 	"fmt"
 	"log"
+	"testing"
 
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
-	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
```
Comment (MrAlias) [inline exporters/stdout/stdouttrace/example_test.go:27]: Do not downgrade the semconv used here.

File: exporters/stdout/stdouttrace/example_test.go [hunk 2]
```diff
@@ -77,11 +78,11 @@ func InstallExportPipeline(ctx context.Context) (func(context.Context) error, er
 	return tracerProvider.Shutdown, nil
 }
 
-func Example() {
+func TestExample(t *testing.T) {
```
Comment (MrAlias) [inline exporters/stdout/stdouttrace/example_test.go:81]: This transforms the example test into a test. The name should not be modified.

https://go.dev/blog/examples

**Review by MrAlias — CHANGES_REQUESTED** [review]

**Comment by MrAlias** [comment]
Please be sure to sign the CLA so these changes can be considered.

File: exporters/stdout/stdouttrace/example_test.go [hunk 3]
```diff
@@ -91,5 +91,8 @@ func Example() {
 		}
 	}()
 
-	log.Println("the answer is", add(ctx, multiply(ctx, multiply(ctx, 2, 2), 10), 2))
+	ctx, ans := multiply(ctx, 2, 2)
+	ctx, ans = multiply(ctx, ans, 10)
+	ctx, ans = add(ctx, ans, 2)
```
Comment (pellared) [inline exporters/stdout/stdouttrace/example_test.go:96]: I find returning a `ctx` pretty unusual (code smell). Why not creating a parent span like:



```suggestion
	ctx, span = tracer.Start(ctx, "Calculation")
	defer span.End()
	ans := multiply(ctx, 2, 2)
	ans = multiply(ctx, ans, 10)
	ans = add(ctx, ans, 2)
```

**Review by pellared — CHANGES_REQUESTED** [review]

**Pull request merged**